features/net/network-socket/SocketAddress.h

```diff
@@ -78,9 +78,10 @@ class SocketAddress {
    
     /** Set the IP address
      *
-     *  @param addr     Null-terminated represention of the IP address
+     *  @param addr     Null-terminated representation of the IP address
+     *  @return         True if address was a valid IP literal
      */
-    void set_ip_address(const char *addr);
+    bool set_ip_address(const char *addr);
```

Comment:
Not sure this is needed. If the `set_ip_address` call fails, the resulting `SocketAddress` will be null. Failure can be checked like so:
``` cpp
SocketAddress address;
address.set_ip_address("1.2.3.4");
if (!address) {
    // handle invalid address
}
```

We should avoid changing existing APIs unless there is evidence that the current API is insufficient or conflicts with the current coding guidelines.